server/engine/src/bad_frames_detector/thresholds.py

```diff
@@ -0,0 +1,436 @@
+import nimblephysics as nimble
+from typing import List, Tuple, Optional, Dict
+from bad_frames_detector.abstract_detector import AbstractDetector
+import json
+import numpy as np
+import os
+import importlib.resources
+
+
+class ThresholdsDetector(AbstractDetector):
```

Comment:
Would you mind adding a brief description of this detector? There is a lot of complicated logic below, so an overview of each step would be very useful for maintainability.

Reply:
Added lots of comments all over the file.